pilot/pkg/model/jwks_resolver.go

```diff
@@ -266,17 +265,35 @@ func (r *JwksResolver) GetPublicKey(issuer string, jwksURI string) (string, erro
 // BuildLocalJwks builds local Jwks by fetching the Jwt Public Key from the URL passed if it is empty.
 func (r *JwksResolver) BuildLocalJwks(jwksURI, jwtIssuer, jwtPubKey string) *envoy_jwt.JwtProvider_LocalJwks {
 	if jwtPubKey == "" {
-		var err error
+		// var err error
 
 		// jwtKeyResolver should never be nil since the function is only called in Discovery Server request processing
 		// workflow, where the JWT key resolver should have already been initialized on server creation.
-		jwtPubKey, err = r.GetPublicKey(jwtIssuer, jwksURI)
-		if err != nil {
-			log.Errorf("Failed to fetch jwt public key from issuer %q, jwks uri %q: %s", jwtIssuer, jwksURI, err)
-			// This is a temporary workaround to reject a request with JWT token by using a fake jwks when istiod failed to fetch it.
-			// TODO(xulingqing): Find a better way to reject the request without using the fake jwks.
+		_, _ = r.GetPublicKey(jwtIssuer, jwksURI)
+		key := jwtKey{issuer: jwtIssuer, jwksURI: jwksURI}
+		if val, found := r.keyEntries.Load(key); found {
+			e := val.(jwtPubKeyEntry)
+			if e.pubKey == "" {
+				log.Infof("no public key present in the cache, adding the key to the cache")
```

Comment:
We lost the comments on why we use the fake jwks. Also the log says adding to cache but I don't see where we do that